src/rabin.rs

```diff
@@ -0,0 +1,95 @@
+use byteorder::{ByteOrder, LittleEndian};
+use digest::{consts::U8, generic_array::GenericArray, FixedOutput, Reset, Update};
+use lazy_static::lazy_static;
+
+const EMPTY: i64 = -4513414715797952619;
+
+lazy_static! {
+    static ref FPTABLE: [i64; 256] = {
+        let mut fp_table: [i64; 256] = [0; 256];
+        for i in 0..256 {
+            let mut fp = i;
+            for _ in 0..8 {
+                fp = (fp as u64 >> 1) as i64 ^ (EMPTY & -(fp & 1));
+            }
+            fp_table[i as usize] = fp
+        }
+        fp_table
+    };
+}
+
+/// Implementation of the Rabin fingerprint algorithm
+#[derive(Clone)]
+pub struct Rabin {
+    result: i64,
+}
+
+impl Default for Rabin {
```

Comment:
What's the reason for not implementing the `Digest` trait? On the [official documentation](https://docs.rs/digest/0.9.0/digest/trait.Digest.html)
> It's a convenience wrapper around Update, FixedOutput, Reset, Clone, and Default traits. It also provides additional convenience methods.

Also, from our definition of fingerprint: https://github.com/flavray/avro-rs/blob/b3edfacced605a9d3229426441a2808176c86120/src/schema.rs#L416

Reply:
digest has an implementation of `Digest` if you implement the traits I implemented:
```
impl<D: Update + FixedOutput + Reset + Clone + Default> Digest for D
```
And that implements a few of the functions for you, e.g. `chain`, `output_size`, `digest`. So it seems easier. 
Also, I was looking at the `Digest` implementation in the `sha2` crate, and that is how they do it.

Reply:
I see. Thanks for the explanation.